argoverse/evaluation/eval_tracking.py

```diff
@@ -104,176 +104,149 @@ def get_distance(x1: Dict[str, np.ndarray], x2: Dict[str, np.ndarray], name: str
         raise NotImplementedError("Not implemented..")
 
 
-def eval_tracks(
+def get_frame_data(
     path_tracker_output_root: _PathLike,
-    path_dataset_root: _PathLike,
+    log_id: str,
+    path_track_data,
+    path_dataset,
+    category: str,
+    diffatt: Optional[str],
     d_min: float,
     d_max: float,
-    out_file: TextIO,
-    centroid_method: str,
-    diffatt: Optional[str],
-    category: str = "VEHICLE",
-) -> None:
-    """Evaluate tracking output.
-
-    Args:
-        path_tracker_output_root: path to tracker output root, containing log_id subdirs
-        path_dataset_root: path to dataset root, containing log_id subdirs
-        d_min: minimum allowed distance range for ground truth and predicted objects,
-            in meters
-        d_max: maximum allowed distance range, as above, in meters
-        out_file: output file object
-        centroid_method: method for ground truth centroid estimation
-        diffatt: difficulty attribute ['easy',  'far', 'fast', 'occ', 'short']. Note that if
-            tracking according to a specific difficulty attribute is computed, then all ground
-            truth annotations not fulfilling that attribute specification are
-            disregarded/dropped out. Since the corresponding track predictions are not dropped
-            out, the number of false positives, false negatives, and MOTA will not be accurate
-            However, `mostly tracked` and `mostly lost` will be accurate.
-        category: such as "VEHICLE" "PEDESTRIAN"
-    """
-    acc_c = mm.MOTAccumulator(auto_id=True)
-    acc_i = mm.MOTAccumulator(auto_id=True)
-    acc_o = mm.MOTAccumulator(auto_id=True)
-
-    ID_gt_all: List[str] = []
-
-    count_all: int = 0
-    if diffatt is not None:
-        import argoverse.evaluation
-
-        pkl_path = os.path.join(os.path.dirname(argoverse.evaluation.__file__), "dict_att_all.pkl")
-        if not os.path.exists(pkl_path):
-            # generate them on the fly
-            logger.info(pkl_path)
-            raise NotImplementedError
-
-        pickle_in = open(pkl_path, "rb")  # open(f"{path_dataset_root}/dict_att_all.pkl","rb")
-        dict_att_all = pickle.load(pickle_in)
-
-    path_datasets = glob.glob(os.path.join(path_dataset_root, "*"))
-    num_total_gt = 0
-
-    for path_dataset in path_datasets:
-
-        log_id = pathlib.Path(path_dataset).name
-        if len(log_id) == 0 or log_id.startswith("_"):
-            continue
-
-        path_tracker_output = os.path.join(path_tracker_output_root, log_id)
-
-        path_track_data = sorted(
-            glob.glob(os.path.join(os.fspath(path_tracker_output), "per_sweep_annotations_amodal", "*"))
+    ID_gt_all: List[str],
+    num_total_gt: int,
+    count_all: int,
+    acc_c,
+    acc_i,
+    acc_o,
+):
```

Comment:
No return type?  Also no types on `acc_c`, `acc_i`, `acc_o` ...

Also, this function has way too many arguments.  It seems like we probably needs a refactor.